packages/just-repo-utils/src/repoInfo.ts

```diff
@@ -0,0 +1,50 @@
+import { getConfigLoader, loadCJson } from './readConfigs';
+import { LernaJson, RushJson, PackageJson } from './interfaces/configTypes';
+import { RepoInfo } from './interfaces/repoInfoTypes';
+import path from 'path';
+import fse from 'fs-extra';
+
+let _repoInfo: RepoInfo | undefined = undefined;
+
+/**
+ * Everyone seems to have one of these, so this is a common implementation that can
+ * be leveraged by different utilities.
+ *
+ * @param cb - callback function to execute at each level.  A true result for the callback will
+ * cancel the walk and return the current path at the time it was cancelled.
+ */
+export function findGitRoot(cb?: (current: string) => boolean | void): string {
+  let cwd = process.cwd();
+  const root = path.parse(cwd).root;
+
+  while (cwd !== root) {
+    if ((cb && cb(cwd)) || fse.existsSync(path.join(cwd, '.git'))) {
+      return cwd;
+    }
+    cwd = path.dirname(cwd);
+  }
+  throw 'No repository root found!';
+}
+
+/**
+ * Retrieve info for the repository.  This will walk up from the current working directory
+ * until it finds the git root and then prepare loaders for various monorepo config files
+ */
+export function repoInfo(): RepoInfo {
```

Comment:
Can we rename this function to something with a verb like "getRepoInfo()" ?

Reply:
Also, please mention that this function works with a module level cache so future users of this function would be careful in how it is used (in tests?)